src/Makefile.in

```diff
@@ -984,10 +984,19 @@ peflags:
 	@echo Securing executables.
 	$(PEFLAGS) $(PROJ)
 
+###############################################################################
+# To use it, run
+# $ make -s windows-tiny-link-to-john _BEST_SIMD=\"\\\"john-avx2.exe\\\"\"
+###############################################################################
+.PHONY: windows-tiny-link-to-john
+windows-tiny-link-to-john:
+	@echo Build a tiny john.exe that calls the best SIMD binary.
+	$(CC) -DMAIN_NAME=$(_BEST_SIMD) symlink.c -o ../run/john.exe
```

Comment:
Maybe we should use the same CFLAGS that we use for john.c. Also, I'd avoid the leading underscore in `_BEST_SIMD`. And maybe call it simply `MAIN_NAME`, for consistency.

Reply:
In fact, I tested this and I noticed it has:
- `-g -O2 -mavx512bw ...`. To me that seems too much. But, it everything seems ok. I can do it, if you really want.

Reply:
We must not be compiling the main john.c with any -m[SIMD] option - that's a bug that we should fix separately. Until we've fixed it, OK, let's not use those same CFLAGS here, but can we hard-code just `-O2` here for now, then?

Also, where do(?) we strip binaries for our packaged Windows builds? Should we use `-s` on commands like this?

Reply:
> We must not be compiling the main john.c with any -m[SIMD] option - that's a bug that we should fix separately. 

It is possible I picked the wrong `xFLAGS`, I'm not sure now.

> Also, where do(?) we strip binaries for our packaged Windows builds?

Well, make CANNOT strip for you. The packager has to run `make strip`.

Reply:
OK. I've just added my proposed changes to here. I think this is ready to merge, but technically we have to wait for CI before we can merge...

Reply:
I added `CFLAGS`.It is ok.

Reply:
Ops. No it is not.

```
Executing 'make windows-tiny-link-to-john MAIN_NAME=john-best.exe '
Build a tiny john.exe that calls the best SIMD binary.
gcc -DAC_BUILT -mavx2 -DJOHN_AVX2 -c -m64 -g -O2 -DARCH_LITTLE_ENDIAN=1  -Wall -Wno-format-overflow -Wno-format-truncation -fno-omit-frame-pointer --param allow-store-data-races=0 -Wno-deprecated-declarations -Wunused-but-set-variable -std=gnu89 -Wdate-time -D_GNU_SOURCE -D__CYGWIN64__ -D__CYGWIN32__  -fopenmp   -DCL_SILENCE_DEPRECATION -DMAIN_NAME=\"john-best.exe\" symlink.c -o ../run/john.exe
strip ../run/john.exe
```

```
.PHONY: windows-tiny-link-to-john
windows-tiny-link-to-john: symlink.c
	@echo Build a tiny john.exe that calls the best SIMD binary.
-	$(CC) -DMAIN_NAME=\"$(_BEST_SIMD)\" symlink.c -o ../run/john.exe
+	$(CC) $(CFLAGS) -DMAIN_NAME=\"$(MAIN_NAME)\" symlink.c -o ../run/john.exe
```

Reply:
Claudio, with your latest force-pushes, you dropped my other changes (such as edits to the comment), and they're now hard to find (I guess `reflog` on your branch could find them). Not to clash again, I'll just ask you what to edit: please add "-O2 -Wall -s " just before "-o", and drop the separate "$(STRIP)" line.

Reply:
https://github.com/magnumripper/JohnTheRipper/commit/429ef2bd40336afab8ad6e50cab0a006dd61afd1

Reply:
Your changes are back in the PR.

Reply:
Thanks! I now think it would also be good to add "-Wall", but this is a minor detail. Feel free to add this and merge, or to merge as-is.